### PR TITLE
Clean code and change API

### PR DESCRIPTION
**Pull request by fsuc**

### CODE REVIEW

File: OCast/Classes/Core/Device/DeviceManager.swift [hunk 1]
```diff
@@ -18,8 +18,29 @@
 
 import Foundation
 
+/// The delegate of a DeviceManager object must adopt the DeviceManagerDelegateDelegate protocol.
 @objc public protocol DeviceManagerDelegate {
-    func deviceDidDisconnect(module: DriverModule, withError error: NSError?)
+
```
Comment (chazemar) [inline OCast/Classes/Core/Device/DeviceManager.swift]: Add extension with default implementation

File: OCast/Classes/Core/Application/Media/MediaController.swift [hunk 2]
```diff
@@ -46,12 +46,9 @@ import Foundation
 public final class MediaController: NSObject, DataStream {
     private static let mediaControllerErrorDomainName = "MediaController"
     
-    private let delegate: MediaControllerDelegate?
+    /// The delegate to receive media events
+    public weak var delegate: MediaControllerDelegate?
     
-    internal init(with delegate: MediaControllerDelegate?) {
```
Comment (chazemar) [inline OCast/Classes/Core/Application/Media/MediaController.swift]: Add default constructor init() with internal visibility

File: OCast/Classes/Core/Application/Media/MediaController.swift [hunk 3]
```diff
@@ -17,22 +17,23 @@
...
-     Gets called when a media status has been received from the web application
-     - Parameter playbackStatus: status information. See `PlaybackStatus` for details.
-     */
-    func didReceiveEvent(playbackStatus: PlaybackStatus)
+    
+    /// Tells the delegate that the mediaController has received a playback status event.
+    ///
+    /// - Parameters:
+    ///   - mediaController: The `MediaController` instance.
+    ///   - playbackStatus: The `PlaybackStatus`object containing playback information.
```
Comment (marcc-orange) [inline OCast/Classes/Core/Application/Media/MediaController.swift]: missing space

File: OCast/Classes/Core/Application/Media/MediaController.swift [hunk 4]
```diff
@@ -17,22 +17,23 @@
...
-    /**
-     Gets called when the media metadata changed.
-     - Parameter metadata: metadata information. See `Metadata` for details.
-     */
-    func didReceiveEvent(metadata: Metadata)
+    /// Tells the delegate that the mediaController has received a metadata event.
+    ///
+    /// - Parameters:
+    ///   - mediaController: The `MediaController` instance.
+    ///   - metadata: The `Metadata`object containing metadata information.
```
Comment (marcc-orange) [inline OCast/Classes/Core/Application/Media/MediaController.swift]: same

File: OCast/Classes/Core/Device/DeviceDiscovery.swift [hunk 5]
```diff
@@ -195,71 +188,46 @@ import CocoaAsyncSocket
...
     }
 
     func sendMSearch() {
-        if let ssdpSocket = ssdpSocket {
-            var tagIndex = 0
-            mSearchTargets.forEach { (target) in
-                let mSearchString = "M-SEARCH * HTTP/1.1\r\nHOST: \(ssdpAddress):\(ssdpPort)\r\nMan: \"ssdp:discover\"\r\nMX:\(mSearchTimeout) \r\nST: \(target)\r\n\r\n"
+        var tagIndex = 0
+        mSearchTargets.forEach { (target) in
+            let mSearchString = "M-SEARCH * HTTP/1.1\r\nHOST: \(ssdpAddress):\(ssdpPort)\r\nMan: \"ssdp:discover\"\r\nMX:\(mSearchTimeout) \r\nST: \(target)\r\n\r\n"
```
Comment (marcc-orange) [inline OCast/Classes/Core/Device/DeviceDiscovery.swift]: Even if case is ignored for HTTP headers, should be more like `MAN:` than `Man:`...

File: OCast/Classes/Core/Device/DeviceDiscovery.swift [hunk 6]
```diff
@@ -195,71 +188,46 @@ import CocoaAsyncSocket
...
-            }
-        }
-
-        if mSearchIdx == Int.max {
-            OCastLog.debug("mSearchIdx reached the maximum allowed value.")
-            currentDevicesIdx.keys.forEach({
-                currentDevicesIdx[$0] = 0
-            })
-            mSearchIdx = 0
+            ssdpSocket.send(mSearchString.data(using: String.Encoding.utf8)!, toHost: ssdpAddress, port: ssdpPort, withTimeout: 1, tag: tagIndex)
```
Comment (marcc-orange) [inline OCast/Classes/Core/Device/DeviceDiscovery.swift]: Just `.utf8`

**Comment by marcc-orange** [comment]
LGTM 👍

**Comment by chazemar** [comment]
nice!

**Pull request merged**